pkg/test/ginkgo/cmd_runsuite.go

```diff
@@ -347,14 +352,37 @@ func (o *GinkgoRunSuiteOptions) Run(suite *TestSuite, clusterConfig *clusterdisc
 		}
 	}
 
-	parallelism := o.Parallelism
-	if parallelism == 0 {
-		parallelism = suite.Parallelism
+	// start with suite value
+	parallelism := suite.Parallelism
+	logrus.Infof("Suite defined parallelism %d", parallelism)
+
+	// adjust based on the number of workers
+	totalNodes, workerNodes, err := getClusterNodeCounts(ctx, restConfig)
+	if err != nil {
+		logrus.Errorf("Failed to get cluster node counts: %v", err)
+	} else {
+		// default to 10 concurrent tests per worker but use the min of that
+		// and the current parallelism value
+		if workerNodes > 0 {
+			workerParallelism := 10 * workerNodes
+			logrus.Infof("Parallelism based on worker node count: %d", workerParallelism)
+			parallelism = min(parallelism, workerParallelism)
+		}
 	}
+
```

Comment:
Could we separate the code that generates the autodl and the code that changes the behavior of the parallelism? Seeing this changed in a PR titled "track run options" is surprising